apps/src/templates/instructions/DocumentationTab.jsx

```diff
@@ -1,26 +1,101 @@
-import React from 'react';
+import React, {useState} from 'react';
 import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
 import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 import {useFetch} from '@cdo/apps/util/useFetch';
+import ProgrammingClassOverview from '../codeDocs/ProgrammingClassOverview';
+import {TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+const DEFAULT_CLASS = 'Java Basics - The main Method';
 
-/**
- * This is a placeholder for the upcoming "Documentation" tab in Javalab.
- * See related task: https://codedotorg.atlassian.net/browse/CSA-361
- */
 const UnconnectedDocumentationTab = function({programmingEnvironment}) {
   const {loading, data, error} = useFetch(
     `/programming_environments/${programmingEnvironment}/get_summary_by_name`,
     null,
     [programmingEnvironment]
   );
 
+  const [selectedClass, setSelectedClass] = useState(null);
+
+  const classMap = {};
+  let firstKey = null;
+  if (data) {
+    for (const category of data) {
```

Comment:
Does this map have to be rebuilt on every render? Could we do this in a `useEffect` callback the first time data is loaded?

Reply:
+1. Also a comment on what this is doing would be super helpful!

Reply:
updated and added a comment